src/main/java/ViewModel/playlistViewModel.java

```diff
@@ -1,4 +1,24 @@
 package ViewModel;
+import Controller.newPlaylistController;
+import InputBoundary.newPlaylistInputBoundary;
+import InputData.newPlaylistInputData;
+import UseCase.createPlaylist;
 
+
+import Presenter.playlistPresenter;
 public class playlistViewModel {
+    private final newPlaylistController newPlaylistController;
+    private final playlistPresenter playlistPresenter;
+
+    public playlistViewModel(){
+        this.playlistPresenter = new playlistPresenter();
+        newPlaylistInputBoundary newPlaylistInputBoundary = new createPlaylist();
+        this.newPlaylistController = new newPlaylistController(newPlaylistInputBoundary, this.playlistPresenter);
```

Comment:
This requires an input data variable, based on the way you wrote newPlaylistController.java. In actuality though, I believe the controller is supposed to create the input data, not get it from the ViewModel. I believe this is why your autograde fails.

Reply:
I don't think you need a newPlaylistInputBoundary either, because IBs are supposed to wrap controllers -- why would you pass a controller to the construction of a new controller?